var/spack/repos/builtin/packages/py-mlflow/package.py

```diff
@@ -0,0 +1,62 @@
+
+from spack.package import *
+
+
+class PyMlflow(PythonPackage):
+    """MLflow: A Platform for ML Development and Productionization."""
+
+    homepage = "https://pypi.org/project/mlflow/"
+    pypi = "mlflow/mlflow-2.0.1.tar.gz"
+
+    version("2.0.1", sha256="7ce6caf3c6acb022d6f5ce8a0995a92be1db524ae16aade1f83da661cdf993de")
+    version("1.17.0", sha256="4898c58899e3101e09e2b37cf5bee7db04c5d73389a56942d3ef5a5e4396799e")
+
+    variant(
+        "skinny",
+        default=False,
+        description="Lightweight MLflow package without SQL storage, server, UI, "
+        "or data science dependencies.",
+    )
+
+    depends_on("python@3.6:", type=("build", "run"), when="@1.17.0:")
+    depends_on("python@3.7:", type=("build", "run"), when="@1.24.0:")
+    depends_on("python@3.8:", type=("build", "run"), when="@2.0.1:")
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-click@7.0:", type=("build", "run"))
+    depends_on("py-cloudpickle", type=("build", "run"))
+    depends_on("py-databricks-cli@0.8.7:", type=("build", "run"))
+    depends_on("py-entrypoints", type=("build", "run"))
+    depends_on("py-gitpython@2.1.0:", type=("build", "run"))
+    depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-protobuf@3.6.0:", type=("build", "run"))
+    depends_on("py-pytz", type=("build", "run"))
+    depends_on("py-requests@2.17.3:", type=("build", "run"))
+    depends_on("py-packaging@:21", type=("build", "run"))
+    depends_on("py-importlib-metadata@3.7.0:4.6.4,4.7.1:5", type=("build", "run"))
+    depends_on("py-sqlparse@0.4.0:0", type=("build", "run"))
+
+    depends_on("py-alembic@:1", type=("build", "run"), when="~skinny")
+    depends_on("py-docker@4.0.0:6", type=("build", "run"), when="~skinny")
+    depends_on("py-flask@:2", type=("build", "run"), when="~skinny")
+    depends_on("py-numpy@:1", type=("build", "run"), when="~skinny")
+    depends_on("py-scipy@:1", type=("build", "run"), when="~skinny")
+    depends_on("py-pandas@:1", type=("build", "run"), when="~skinny")
+    depends_on("py-querystring-parser@:1", type=("build", "run"), when="~skinny")
+    depends_on("py-sqlalchemy@1.4.0:1", type=("build", "run"), when="~skinny")
+    for platform in ["linux", "darwin", "cray"]:
+        depends_on("py-gunicorn@:20", type=("build", "run"), when=f"~skinny platform={platform}")
+    depends_on("py-waitress@:2", type=("build", "run"), when="os=windows ~skinny")
+    depends_on("py-scikit-learn@:1", type=("build", "run"), when="~skinny")
+    depends_on("py-pyarrow@4.0.0:10", type=("build", "run"), when="~skinny")
+    depends_on("py-shap@0.40:0", type=("build", "run"), when="~skinny")
+    depends_on("py-markdown@3.3:3", type=("build", "run"), when="~skinny")
+    for platform in ["linux", "darwin", "cray"]:
+        depends_on("py-jinja2@2.11:3", type=("build", "run"), when=f"~skinny platform={platform}")
+    depends_on("py-jinja2@3.0:3", type=("build", "run"), when="os=windows ~skinny")
```

Comment:
```suggestion
    depends_on("py-waitress@:2", type=("build", "run"), when="platform=windows ~skinny")
    depends_on("py-scikit-learn@:1", type=("build", "run"), when="~skinny")
    depends_on("py-pyarrow@4.0.0:10", type=("build", "run"), when="~skinny")
    depends_on("py-shap@0.40:0", type=("build", "run"), when="~skinny")
    depends_on("py-markdown@3.3:3", type=("build", "run"), when="~skinny")
    for platform in ["linux", "darwin", "cray"]:
        depends_on("py-jinja2@2.11:3", type=("build", "run"), when=f"~skinny platform={platform}")
    depends_on("py-jinja2@3.0:3", type=("build", "run"), when="platform=windows ~skinny")
```
Windows is a platform just like linux/macos/cray